{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4011e79c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'spatstat.core':\n",
      "  method          from\n",
      "  formula.glmmPQL MASS\n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "Attaching sp\n",
      "\n",
      "Loading required package: SingleCellExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:sp’:\n",
      "\n",
      "    %over%\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SummarizedExperiment’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "Loading required package: scuttle\n",
      "\n",
      "Loading required package: ggplot2\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: Rcpp\n",
      "\n"
     ]
    }
   ],
   "source": [
    ".libPaths(c(\"/home/data/t060324/R/x86_64-pc-linux-gnu-library/4.1\",\"/home/data/refdir/Rlib\"))\n",
    "library(scibet)\n",
    "library(Seurat)\n",
    "library(scater)\n",
    "library(scran)\n",
    "library(dplyr)\n",
    "library(Matrix)\n",
    "library(cowplot)\n",
    "library(ggplot2)\n",
    "library(harmony)\n",
    "setwd('/home/data/t060324/jobs/aimin/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a66a4068",
   "metadata": {},
   "outputs": [],
   "source": [
    "sc_data <- readRDS(file='./output/1.sangleCell_pre/OV_GSE165897_final.RDS')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "cfd7c88d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 methods overwritten by 'registry':\n",
      "  method               from \n",
      "  print.registry_field proxy\n",
      "  print.registry_entry proxy\n",
      "\n",
      "Warning message:\n",
      "“replacing previous import ‘RcppML::nmf’ by ‘NMF::nmf’ when loading ‘CARD’”\n"
     ]
    }
   ],
   "source": [
    "library(CARD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "2efb5897",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6 x 6 sparse Matrix of class \"dgCMatrix\"\n",
       "              AAACGGGCACTCTGTC-EOC443_iOme1_1 AAACGGGTCCAAACTG-EOC443_iOme1_1\n",
       "RP11-34P13.7                                .                               .\n",
       "RP11-34P13.8                                .                               .\n",
       "CICP27                                      .                               .\n",
       "RP11-34P13.15                               .                               .\n",
       "RP11-34P13.13                               .                               .\n",
       "FO538757.1                                  .                               .\n",
       "              AAAGATGGTCCATGAT-EOC443_iOme1_1 AAAGTAGGTGGCGAAT-EOC443_iOme1_1\n",
       "RP11-34P13.7                                .                               .\n",
       "RP11-34P13.8                                .                               .\n",
       "CICP27                                      .                               .\n",
       "RP11-34P13.15                               .                               .\n",
       "RP11-34P13.13                               .                               .\n",
       "FO538757.1                                  .                               .\n",
       "              AAAGTAGTCAGAGGTG-EOC443_iOme1_1 AACCATGAGTGTCCCG-EOC443_iOme1_1\n",
       "RP11-34P13.7                                .                               .\n",
       "RP11-34P13.8                                .                               .\n",
       "CICP27                                      .                               .\n",
       "RP11-34P13.15                               .                               .\n",
       "RP11-34P13.13                               .                               .\n",
       "FO538757.1                                  .                               ."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### scRNA count\n",
    "sc_count <- GetAssayData(sc_data,assay = 'RNA',slot = 'counts')\n",
    "sc_count[1:6,1:6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "ea1640fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cellID</th><th scope=col>cellType_auto</th><th scope=col>patient_id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>AAACGGGCACTCTGTC-EOC443_iOme1_1</th><td>AAACGGGCACTCTGTC-EOC443_iOme1_1</td><td>B     </td><td>EOC443</td></tr>\n",
       "\t<tr><th scope=row>AAACGGGTCCAAACTG-EOC443_iOme1_1</th><td>AAACGGGTCCAAACTG-EOC443_iOme1_1</td><td>B     </td><td>EOC443</td></tr>\n",
       "\t<tr><th scope=row>AAAGATGGTCCATGAT-EOC443_iOme1_1</th><td>AAAGATGGTCCATGAT-EOC443_iOme1_1</td><td>B     </td><td>EOC443</td></tr>\n",
       "\t<tr><th scope=row>AAAGTAGGTGGCGAAT-EOC443_iOme1_1</th><td>AAAGTAGGTGGCGAAT-EOC443_iOme1_1</td><td>B     </td><td>EOC443</td></tr>\n",
       "\t<tr><th scope=row>AAAGTAGTCAGAGGTG-EOC443_iOme1_1</th><td>AAAGTAGTCAGAGGTG-EOC443_iOme1_1</td><td>plasma</td><td>EOC443</td></tr>\n",
       "\t<tr><th scope=row>AACCATGAGTGTCCCG-EOC443_iOme1_1</th><td>AACCATGAGTGTCCCG-EOC443_iOme1_1</td><td>plasma</td><td>EOC443</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & cellID & cellType\\_auto & patient\\_id\\\\\n",
       "  & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tAAACGGGCACTCTGTC-EOC443\\_iOme1\\_1 & AAACGGGCACTCTGTC-EOC443\\_iOme1\\_1 & B      & EOC443\\\\\n",
       "\tAAACGGGTCCAAACTG-EOC443\\_iOme1\\_1 & AAACGGGTCCAAACTG-EOC443\\_iOme1\\_1 & B      & EOC443\\\\\n",
       "\tAAAGATGGTCCATGAT-EOC443\\_iOme1\\_1 & AAAGATGGTCCATGAT-EOC443\\_iOme1\\_1 & B      & EOC443\\\\\n",
       "\tAAAGTAGGTGGCGAAT-EOC443\\_iOme1\\_1 & AAAGTAGGTGGCGAAT-EOC443\\_iOme1\\_1 & B      & EOC443\\\\\n",
       "\tAAAGTAGTCAGAGGTG-EOC443\\_iOme1\\_1 & AAAGTAGTCAGAGGTG-EOC443\\_iOme1\\_1 & plasma & EOC443\\\\\n",
       "\tAACCATGAGTGTCCCG-EOC443\\_iOme1\\_1 & AACCATGAGTGTCCCG-EOC443\\_iOme1\\_1 & plasma & EOC443\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | cellID &lt;chr&gt; | cellType_auto &lt;chr&gt; | patient_id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| AAACGGGCACTCTGTC-EOC443_iOme1_1 | AAACGGGCACTCTGTC-EOC443_iOme1_1 | B      | EOC443 |\n",
       "| AAACGGGTCCAAACTG-EOC443_iOme1_1 | AAACGGGTCCAAACTG-EOC443_iOme1_1 | B      | EOC443 |\n",
       "| AAAGATGGTCCATGAT-EOC443_iOme1_1 | AAAGATGGTCCATGAT-EOC443_iOme1_1 | B      | EOC443 |\n",
       "| AAAGTAGGTGGCGAAT-EOC443_iOme1_1 | AAAGTAGGTGGCGAAT-EOC443_iOme1_1 | B      | EOC443 |\n",
       "| AAAGTAGTCAGAGGTG-EOC443_iOme1_1 | AAAGTAGTCAGAGGTG-EOC443_iOme1_1 | plasma | EOC443 |\n",
       "| AACCATGAGTGTCCCG-EOC443_iOme1_1 | AACCATGAGTGTCCCG-EOC443_iOme1_1 | plasma | EOC443 |\n",
       "\n"
      ],
      "text/plain": [
       "                                cellID                          cellType_auto\n",
       "AAACGGGCACTCTGTC-EOC443_iOme1_1 AAACGGGCACTCTGTC-EOC443_iOme1_1 B            \n",
       "AAACGGGTCCAAACTG-EOC443_iOme1_1 AAACGGGTCCAAACTG-EOC443_iOme1_1 B            \n",
       "AAAGATGGTCCATGAT-EOC443_iOme1_1 AAAGATGGTCCATGAT-EOC443_iOme1_1 B            \n",
       "AAAGTAGGTGGCGAAT-EOC443_iOme1_1 AAAGTAGGTGGCGAAT-EOC443_iOme1_1 B            \n",
       "AAAGTAGTCAGAGGTG-EOC443_iOme1_1 AAAGTAGTCAGAGGTG-EOC443_iOme1_1 plasma       \n",
       "AACCATGAGTGTCCCG-EOC443_iOme1_1 AACCATGAGTGTCCCG-EOC443_iOme1_1 plasma       \n",
       "                                patient_id\n",
       "AAACGGGCACTCTGTC-EOC443_iOme1_1 EOC443    \n",
       "AAACGGGTCCAAACTG-EOC443_iOme1_1 EOC443    \n",
       "AAAGATGGTCCATGAT-EOC443_iOme1_1 EOC443    \n",
       "AAAGTAGGTGGCGAAT-EOC443_iOme1_1 EOC443    \n",
       "AAAGTAGTCAGAGGTG-EOC443_iOme1_1 EOC443    \n",
       "AACCATGAGTGTCCCG-EOC443_iOme1_1 EOC443    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### sc mate data\n",
    "sc_data$cellID <- rownames(sc_data@meta.data)\n",
    "sc_meta <- sc_data@meta.data[,c('cellID','cellType_auto','patient_id')]\n",
    "head(sc_meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "d3a24348",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>GSM6506110</dt><dd>'GSM6506110_ST.RDS'</dd><dt>GSM6506111</dt><dd>'GSM6506111_ST.RDS'</dd><dt>GSM6506112</dt><dd>'GSM6506112_ST.RDS'</dd><dt>GSM6506113</dt><dd>'GSM6506113_ST.RDS'</dd><dt>GSM6506114</dt><dd>'GSM6506114_ST.RDS'</dd><dt>GSM6506115</dt><dd>'GSM6506115_ST.RDS'</dd><dt>GSM6506116</dt><dd>'GSM6506116_ST.RDS'</dd><dt>GSM6506117</dt><dd>'GSM6506117_ST.RDS'</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[GSM6506110] 'GSM6506110\\_ST.RDS'\n",
       "\\item[GSM6506111] 'GSM6506111\\_ST.RDS'\n",
       "\\item[GSM6506112] 'GSM6506112\\_ST.RDS'\n",
       "\\item[GSM6506113] 'GSM6506113\\_ST.RDS'\n",
       "\\item[GSM6506114] 'GSM6506114\\_ST.RDS'\n",
       "\\item[GSM6506115] 'GSM6506115\\_ST.RDS'\n",
       "\\item[GSM6506116] 'GSM6506116\\_ST.RDS'\n",
       "\\item[GSM6506117] 'GSM6506117\\_ST.RDS'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "GSM6506110\n",
       ":   'GSM6506110_ST.RDS'GSM6506111\n",
       ":   'GSM6506111_ST.RDS'GSM6506112\n",
       ":   'GSM6506112_ST.RDS'GSM6506113\n",
       ":   'GSM6506113_ST.RDS'GSM6506114\n",
       ":   'GSM6506114_ST.RDS'GSM6506115\n",
       ":   'GSM6506115_ST.RDS'GSM6506116\n",
       ":   'GSM6506116_ST.RDS'GSM6506117\n",
       ":   'GSM6506117_ST.RDS'\n",
       "\n"
      ],
      "text/plain": [
       "         GSM6506110          GSM6506111          GSM6506112          GSM6506113 \n",
       "\"GSM6506110_ST.RDS\" \"GSM6506111_ST.RDS\" \"GSM6506112_ST.RDS\" \"GSM6506113_ST.RDS\" \n",
       "         GSM6506114          GSM6506115          GSM6506116          GSM6506117 \n",
       "\"GSM6506114_ST.RDS\" \"GSM6506115_ST.RDS\" \"GSM6506116_ST.RDS\" \"GSM6506117_ST.RDS\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sample <- list.files(path = './data/GSE211956_RAW/',pattern = '_ST.RDS')\n",
    "names(sample) <- substr(sample,1,10)\n",
    "sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "fbcf34c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: nnls\n",
      "\n",
      "\n",
      "Attaching package: ‘wrMisc’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    colSds, rowSds\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colSds, rowSds\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n",
      "## QC on scRNASeq dataset! ...\n",
      "## QC on spatially-resolved dataset! ...\n",
      "## create reference matrix from scRNASeq...\n",
      "## Select Informative Genes! ...\n",
      "## Deconvolution Starts! ...\n",
      "## Deconvolution Finish! ...\n"
     ]
    }
   ],
   "source": [
    "for(samplefile in names(sample)){\n",
    "ST_obj <- readRDS(file = paste0('./data/GSE211956_RAW/',sample[samplefile]))\n",
    "spatial_count <- GetAssayData(ST_obj,assay = 'Spatial',slot = 'counts')\n",
    "spatial_location <- ST_obj@images$image@coordinates[,c('row','col')]\n",
    "colnames(spatial_location) <- c('x','y')\n",
    "CARD_obj = createCARDObject(\n",
    "\tsc_count = sc_count,\n",
    "\tsc_meta = sc_meta,\n",
    "\tspatial_count = spatial_count,\n",
    "\tspatial_location = spatial_location,\n",
    "\tct.varname = \"cellType_auto\",\n",
    "\tct.select = unique(sc_meta$cellType_auto),\n",
    "\tsample.varname = \"patient_id\",\n",
    "\tminCountGene = 100,\n",
    "\tminCountSpot = 5) \n",
    "CARD_obj = CARD_deconvolution(CARD_object = CARD_obj)\n",
    "tmp <- CARD_obj@Proportion_CARD[match(rownames(ST_obj@meta.data),rownames(CARD_obj@Proportion_CARD)),]\n",
    "rownames(tmp) <- rownames(ST_obj@meta.data)\n",
    "ST_obj@meta.data <- cbind(ST_obj@meta.data,tmp)\n",
    "saveRDS(ST_obj,file = paste0('./data/GSE211956_RAW/',sample[samplefile]))\n",
    "saveRDS(CARD_obj,file = paste0('./data/GSE211956_RAW/',samplefile,'_ST_CARD.RDS'))\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
